### ISSUE TITLE
az login fails with SAML token signature validation failure

### ISSUE THREAD

**Issue by swails** [Azure/azure-cli#6287]
---

### Environment summary

Install Method (e.g. pip, interactive script, apt-get, Docker, MSI, edge build) / CLI version (`az --version`) / OS version / Shell Type (e.g. bash, cmd.exe, Bash on Windows)

Windows 10, Powershell, Python 3.5.5 installed from Continuum IO (Miniconda)

```
azure-cli (2.0.31)

acr (2.0.23)
acs (2.0.31)
advisor (0.5.1)
appservice (0.1.31)
backup (1.1.1)
batch (3.2.0)
batchai (0.2.0)
billing (0.1.8)
cdn (0.0.14)
cloud (2.0.13)
cognitiveservices (0.1.12)
command-modules-nspkg (2.0.1)
configure (2.0.15)
consumption (0.3.0)
container (0.1.22)
core (2.0.31)
cosmosdb (0.1.20)
dla (0.0.19)
dls (0.0.21)
eventgrid (0.1.12)
eventhubs (0.1.2)
extension (0.0.12)
feedback (2.1.1)
find (0.2.9)
interactive (0.3.19)
iot (0.1.19)
keyvault (2.0.21)
lab (0.0.21)
monitor (0.1.5)
network (2.0.28)
nspkg (3.0.2)
profile (2.0.22)
rdbms (0.2.1)
redis (0.2.12)
reservations (0.1.2)
resource (2.0.27)
role (2.0.22)
servicebus (0.1.2)
servicefabric (0.0.12)
sql (2.0.25)
storage (2.0.31)
vm (2.0.30)

Python location 'C:\Users\jswails\AppData\Local\Continuum\Miniconda3\python.exe'
Extensions directory 'C:\Users\jswails\.azure\cliextensions'

Python (Windows) 3.5.5 |Anaconda, Inc.| (default, Apr  7 2018, 04:52:34) [MSC v.1900 64 bit (AMD64)]

Legal docs and information: aka.ms/AzureCliLegal
```

When I type `az login --username <my_username> --password <my_password>`, I get the following error:

```
 az login -u jswails@MYDOMAIN.com
Password:
Get Token request returned http error: 400 and server response: {"error":"invalid_grant","error_description":"AADSTS70002: Error validating credentials. AADSTS50008: SAML token is invalid. AADSTS50006: The element with ID '_877ed6b2-409b-45a9-9cce-216b61d460b3' was either unsigned or the signature was invalid.\r\nTrace ID: edabfd9e-2b6e-41cf-87c0-b3d59ce90000\r\nCorrelation ID: 43366876-8bb2-4bb5-a76a-93a77f7ce525\r\nTimestamp: 2018-05-03 20:13:42Z","error_codes":[70002,50008,50006],"timestamp":"2018-05-03 20:13:42Z","trace_id":"edabfd9e-2b6e-41cf-87c0-b3d59ce90000","correlation_id":"43366876-8bb2-4bb5-a76a-93a77f7ce525"}
```

Some background information:

* This command used to work.  It stopped working right around the time our organization made a change to our SSO applications
* This command *still* works on machines in the organization that have never installed/used `az` before (i.e., if I install az and run this command, it works).  But it fails for every machine that has already installed and used "az" before our SSO settings updates (for every account that tries)
* Rebooting the computer does nothing
* Wiping the ~/.azure directory does nothing
* Uninstalling and reinstalling the Azure CLI does nothing

It looks like when our SSO settings were updated, there was a cert change that's not being picked up by my machine, and the new private cert on our login server isn't able to verify the signature on the SAML token signed by my machine.

So I guess my question is: where can I find this certificate to delete it??  I have been staring at this forever, but am at a complete loss.  The powershell cmdlets for logging into Azure continue to work, but az does not.

**Comment by yugangw-msft** [comment]
@swails, I don't think there are any local certificates involved to get the login work, everything are supposed to happen on the wire through the ws-trust protocol.
To isolate the issue, can you cross check 2 things?
    1. For machines where the command works, was the same user account used?
    2. If yes to the first one, can you use "--debug" flag and send me the trace for both the working and nonworking case? I can be reached at yugangw at microsoft dot com

**Comment by swails** [comment]
Hi @yugangw-msft -- yes, this happens for the same account.  Perhaps the (invalid) cached cert is stored on the auth server itself?

I've emailed you the logs privately.

Thanks for looking into this!

**Comment by yugangw-msft** [comment]
Per trace, it is an [ADAL bug](https://github.com/AzureAD/azure-activedirectory-library-for-python/issues/80) which has been fixed since 0.5.1. To get CLI work, either upgrade to newer CLI which uses new ADAL or directly upgrade ADAL (if pip was used to acquire azure-cli)
